src/utils/utils.cairo

```diff
@@ -812,4 +819,34 @@ namespace Helpers {
             data_len=data_len - 1, data=data + 1, bytes_len=bytes_len + 4, bytes=res
         );
     }
+
+    // Computes x * y % p for Uint256 numbers via fast modular exponentiation algorithm.
+    // Time complexity is log_2(y).
+    // Loop is implemented via uint256_expmod_recursive_call() function.
+    func uint256_expmod{range_check_ptr: felt}(x: Uint256, y: Uint256, p: Uint256) -> (remainder: Uint256) {
+        alloc_locals;
+        let res = Uint256(low=1,high=0);
+        let (r_x,r_y,r_res) = uint256_expmod_recursive_call(x,y,res,p);
+        let (quotient,remainder) = uint256_unsigned_div_rem(r_res,p);
+        return (remainder=remainder);
+    }
+
+    func uint256_expmod_recursive_call{range_check_ptr: felt}(x: Uint256, y: Uint256, res: Uint256, p: Uint256) -> (r_x: Uint256, r_y: Uint256, r_res: Uint256) {
+        alloc_locals;
+        let (is_greater_than_zero) = uint256_lt(Uint256(low=0, high=0),y);
+        if ((is_greater_than_zero) == 0) {
+            return (r_x=x,r_y=y,r_res=res);
+        }
+
+        let (quotient, remainder) = uint256_unsigned_div_rem(y,Uint256(low=2, high=0));
+        let (is_equal_to_one) = uint256_eq(remainder,Uint256(low=1, high=0));
+        if ((is_equal_to_one) == 0) {
+            let (x_res_quotient,  x_res_quotient_high,  x_res_remainder) = uint256_mul_div_mod(x,x,p);
+            return uint256_expmod_recursive_call(x=x_res_remainder,y=quotient,res=res,p=p);
+        } else {
+            let (x_res_res_quotient, x_res_res_quotient_high, x_res_res_remainder) = uint256_mul_div_mod(res,x,p);
+            let (x_res_quotient,  x_res_quotient_high,  x_res_remainder) = uint256_mul_div_mod(x,x,p);
+            return uint256_expmod_recursive_call(x=x_res_remainder,y=quotient,res=x_res_res_remainder,p=p);
```

Comment:
can't we use the `if/else` only for setting the first param `a` of the `uint256_mul_div_mod` like

```
if (is_equal_to_one == 0) {
            tempvar a = x;
        } else {
            tempvar a = res;
}
let (x_res_quotient,  x_res_quotient_high,  x_res_remainder) = uint256_mul_div_mod(a,x,p);
return uint256_expmod_recursive_call(x=x_res_remainder,y=quotient,res=x_res_res_remainder,p=p);
```